Makefile

```diff
@@ -612,27 +621,36 @@ $(HELM):
 $(HOME)/.helm:
 	$(HELM) init --client-only
 
+.PHONY: helm-repo-add
+
+helm-repo-add:
+	$(HELM) repo add istio.io ${ISTIO_HELM_REPO}
+
 # create istio-remote.yaml
-istio-remote.yaml: $(HELM) $(HOME)/.helm
+istio-remote.yaml: $(HELM) $(HOME)/.helm helm-repo-add
 	cat install/kubernetes/namespace.yaml > install/kubernetes/$@
 	$(HELM) dep update --skip-refresh install/kubernetes/helm/istio-remote
 	$(HELM) template --name=istio --namespace=istio-system \
+		--set istio_cni.enabled=${ENABLE_ISTIO_CNI} \
+		${EXTRA_HELM_SETTINGS} \
 		install/kubernetes/helm/istio-remote >> install/kubernetes/$@
 
 # creates istio.yaml istio-auth.yaml istio-one-namespace.yaml istio-one-namespace-auth.yaml
 # Ensure that values-$filename is present in install/kubernetes/helm/istio
-isti%.yaml: $(HELM) $(HOME)/.helm
+isti%.yaml: $(HELM) $(HOME)/.helm helm-repo-add
 	$(HELM) dep update --skip-refresh install/kubernetes/helm/istio
```

Comment:
i'm not certain if `--skip-refresh` will work for your use case or even what it does now that I've been away from my computer for a week or so :) . Be prepared to modify this code a bit as time passes.

Reply:
ok ... I'll do a bit of research but it seems ok so far